api/v1alpha1/gitopsservice_types.go

```diff
@@ -32,6 +32,31 @@ type GitopsServiceSpec struct {
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
 	// NodeSelector is a map of key value pairs used for node selection in the default workloads
 	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// ConsolePlugin defines the Resource configuration for the Console Plugin components
+	ConsolePlugin ConsolePluginResourceStruct `json:"consolePlugin,omitempty"`
+}
+
+// ConsolePluginResourceStruct defines the resource configuration for the Console Plugin components
+type ConsolePluginResourceStruct struct {
```

Comment:
Better to name it as `ConsolePluginStruct` as it will help us to extend for other use cases apart from resources.

Reply:
Also can we use pointer to struct, instead of using the struct directly.  similar to the one in argocd operator https://github.com/argoproj-labs/argocd-operator/blob/7dfc7c55cbbfef840097072b6d8e1e55d7040cd1/api/v1beta1/argocd_types.go#L834